classes/class04/homework/routes/base.js

```diff
@@ -0,0 +1,99 @@
+var schema = require('../models/schema');
+
+exports.home = function (req, res) {
+    res.render('index');
+};
+
+exports.ingredients = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        res.render('ingredients', {ingredients: data});
+    })
+}
+
+exports.order = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        if (err) {
+            console.log("Failed to get list of ingredients");
+            res.status(500, {error: "Failed to get list of ingredients " + err});
+            return;
+        }
+        res.render('order', {ingredients: data});
+    });
+}
+
+exports.kitchen = function (req, res) {
+    schema.Order.find({}, function (err, data) {
+        res.render("kitchen", {orders: data});
+    });
+}
+
+exports.addOrder = function (req, res) {
+    var ingredients = [];
+    for (var key in req.body) {
+        if (key && key != "total_cost" && key != "name" && key != "ingredientStr" && key != req.body.hasOwnProperty(key) && req.body[key]) {
+            ingredients.push(key);
+        }
+    }
+    schema.Order({
+          name: req.body.name
+        , ingredients: ingredients
+        , ingredientStr: req.body.ingredientStr
+        , totalCost: req.body.totalCost
+    }).save(function (err) {
+        if (err) {
+            console.log("Error saving order");
+            res.status(500, {error: "Error saving order " + err});
+
+        }
+        console.log(req.body.name);
+        res.render("ordered", {orderName: req.body.name});
+    });
+}
+
+exports.doneOrder = function (req, res) {
+    schema.Order.find({_id: req.body.id}).remove().exec(function (err) {
+        console.log("Could not delete order");
+        res.status(500, {error: "Could not delete order " + err});
```

Comment:
Not strictly necessary, but often a good idea to send some sort of "success" message back even if the client isn't asking for any data.